Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -373,6 +375,14 @@ protected void onResume()
   {
     super.onResume();
 
+    // Only android 9+ support this feature.
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P &&
+            BooleanSetting.MAIN_EXPAND_TO_CUTOUT_AREA.getBoolean(mSettings))
+    {
+      getWindow().getAttributes().layoutInDisplayCutoutMode =
+              WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES;
+    }
```

Comment:
Here is how to fix the code to make it work when returning from the settings:

```suggestion
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P)
    {
      WindowManager.LayoutParams attributes = getWindow().getAttributes();
      
      attributes.layoutInDisplayCutoutMode =
              BooleanSetting.MAIN_EXPAND_TO_CUTOUT_AREA.getBoolean(mSettings) ?
                      WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES :
                      WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_NEVER;
      
      getWindow().setAttributes(attributes);
    }
```

Reply:
done, thanks!